src/main/java/net/william278/velocitab/api/PluginMessageAPI.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * This file is part of Velocitab, licensed under the Apache License 2.0.
+ *
+ *  Copyright (c) William278 <will27528@gmail.com>
+ *  Copyright (c) contributors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package net.william278.velocitab.api;
+
+import com.google.common.collect.Sets;
+import com.velocitypowered.api.event.connection.PluginMessageEvent;
+import com.velocitypowered.api.proxy.Player;
+import com.velocitypowered.api.proxy.ServerConnection;
+import com.velocitypowered.api.proxy.messages.MinecraftChannelIdentifier;
+import net.william278.velocitab.Velocitab;
+import net.william278.velocitab.packet.UpdateTeamsPacket;
+import net.william278.velocitab.player.TabPlayer;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.Set;
+
+public class PluginMessageAPI {
+
+    private final Velocitab plugin;
+    private final Set<MinecraftChannelIdentifier> channels;
+
+    public PluginMessageAPI(@NotNull Velocitab plugin) {
+        this.plugin = plugin;
+        this.channels = Sets.newConcurrentHashSet();
+    }
+
+    public void registerChannel() {
+        Arrays.stream(PluginMessageRequest.values())
+                .map(PluginMessageRequest::name)
+                .forEach(request -> {
+                    final String channelName = request.toLowerCase();
```

Comment:
can we do `toLowerCase(Locale.ENGLISH)`? - and perhapse `Map` to a MinecraftChannelIdentiifer` before doing `eorEach`